Apply this patch if you are using x-symbol-beta and preview-latex is
slow.

--- /tmp/x-symbol-beta/lisp/x-symbol-tex.el	Thu Mar  7 18:03:30 2002
+++ /tmp/x-symbol-tex.el	Fri Mar 29 17:51:04 2002
@@ -886,12 +886,38 @@
 ;;;  AucTeX
 ;;;===========================================================================
 
+(defvar x-symbol-hash-file "")
+(defvar x-symbol-hash-buffer nil)
+(defvar x-symbol-hash-line nil)
+(defvar x-symbol-hash-point nil)
+(defvar x-symbol-hash-tick nil)
+
 (defun x-symbol-tex-error-location ()
   (unless (string= string " ")
     (save-excursion
-      (set-buffer (find-file-noselect file))
+      (if (string= file x-symbol-hash-file)
+	  (set-buffer x-symbol-hash-buffer)
+	(set-buffer (find-file-noselect (setq x-symbol-hash-file file)))
+	(setq x-symbol-hash-buffer (current-buffer))
+	(setq x-symbol-hash-tick nil))
       (when (and x-symbol-mode x-symbol-language)
-	(goto-line (+ offset line))
+	(if (and (eq (current-buffer) x-symbol-hash-buffer)
+		 (eq (buffer-modified-tick) x-symbol-hash-tick)
+		 (<= x-symbol-hash-line (+ offset line)))
+	    (progn
+	      (goto-char x-symbol-hash-point)
+	      (if (/= x-symbol-hash-line (+ offset line))
+		  (if (eq selective-display t)
+		      (re-search-forward "[\n\C-m]" nil
+					 'end
+					 (- (+ offset line)
+					    x-symbol-hash-line))
+		    (forward-line (- (+ offset line) x-symbol-hash-line)))))
+	  (goto-line (+ offset line)))
+	(setq x-symbol-hash-buffer (current-buffer)
+	      x-symbol-hash-line (+ offset line)
+	      x-symbol-hash-point (point)
+	      x-symbol-hash-tick (buffer-modified-tick))
 	(let ((coding (if x-symbol-8bits
 			  (or x-symbol-coding x-symbol-default-coding)
 			t))
@@ -918,7 +944,7 @@
 	    ;; undo
 	    (let ((found (and (search-forward string nil t)
 			      (not (bolp))
-			      (copy-marker (point) t)))
+			      (copy-marker (point) nil)))
 		  (tail buffer-undo-list))
 	      (setq buffer-undo-list t)
 	      (while tail (setq tail (primitive-undo (length tail) tail)))
